test/@runtime.mock.js

```diff
@@ -0,0 +1,37 @@
+const { OSGIServices } = require('./openhab.mock');
+
+jest.mock('@runtime', () => ({
+  DateTimeType: jest.fn(),
+  DecimalType: jest.fn(),
+  StringType: jest.fn(),
+  QuantityType: jest.fn()
+}), { virtual: true });
+
+jest.mock('@runtime/Defaults', () => ({}), { virtual: true });
+
+// Do NOT set the mock implementation here, because some tests need to overwrite them. Instead, default initialize them.
+jest.mock('@runtime/osgi', () => ({
+  bundleContext: {
+    getServiceReference: jest.fn(),
+    getService: jest.fn(),
+    getAllServiceReferences: jest.fn(),
+    registerService: jest.fn()
+  },
+  lifecycle: {
+    addDisposeHook: jest.fn()
+  }
+}), { virtual: true });
+
+const { bundleContext } = require('@runtime/osgi');
```

Comment:
I would try to keep the mocks in the setup files as common and simple as possible and to mock as nearby to the requiring tests as possible. In this particular case (the tests in `triggers.spec.ts`), the `@runtime` module doesn't have to be mocked. The usage of the `Item` type from `items/items.js` pulls in this dependency, which is easy to cut with this mock (in `triggers.spec.ts`):

```js
jest.mock('../items', () => ({ Item: class {} }));
```

With this change, you can skip all the mock implementations in `bundleContext` in this file (beyond this line), the mocking of `OSGIServices` and keep the mock of the `@runtime` module cleaner.

To better explain my thoughts, I've created a pr to your branch: florian-h05/openhab-js#1.

Reply:
@sfriedle  Thanks for your help, I've commited your changes. Could you please re-review (possibly also the other changes to JS files) and then approve if everything is fine?